src/controller/getDreamControllerById.js

```diff
@@ -0,0 +1,47 @@
+import Questionnaire from '../models/dream';
+const mongoose = require('mongoose');
+
+function isNumber(posibleId){
+	let id = parseInt(posibleId, 10);
+	if (isNaN(id)) {
+		return false;
+	}
+	return true;
+}
+
+function idVerification(possibleId) {
+    const id = parseInt(possibleId, 10);
+    if (!isNumber(id)){
```

Comment:
I think you could replace `!isNumber(id)` with `isNaN(id)` and remove the isNumber() function

Reply:
After some research i found that both methods can cause errors because of the way they are implemented. 
`parseInt()` can parse invalid strings in numbers, for example: `parseInt("56ex")` is converted into `56` which should not occur. 
`isNaN()` can fail in some cases where javascript behaves strangely, for example: `isNaN(" ")` returns false... Why? Because javascript casts the empty space`" "` as a `0`, then `isNaN()` is tricked. Another one could be `isNaN(true)` that is the equivalent to say `isNaN(1)`
For those reasons i found that the best way in this case in particular is using `isNaN()` but with an if before, checking if there is an empty space. Due to the nature of the app it would never be possible to evaluate booleans like `true` and `false` because they will come to the function as strings, that would be `"true"` and `"false"`